opentsdb/src/main/java/net/opentsdb/aura/metrics/storage/AerospikeBatchQueryNode.java

```diff
@@ -179,6 +203,22 @@ public AerospikeBatchQueryNode(final AerospikeBatchSourceFactory factory,
                                         .setArraySize(downsampleConfig.intervals())
                                         //.setInfectiousNaN(groupByConfig.getInfectiousNan()) // TODO - fix GB nan
                                         .build();
+                        String agg = downsampleConfig.getAggregator();
+                        useRollupData = aggsInStore.contains(agg);
+                        if (useRollupData) {
+                          AggregatorType aggregatorType = AggregatorType.valueOf(agg);
+                          this.rollupAggId = aggregatorType.getId();
+                          this.rollupAggOrdinal = aggregatorType.getOrdinal();
+                          int dsInterval = (int) downsampleConfig.interval().get(ChronoUnit.SECONDS);
+                          int rollupInterval = this.rollupInterval.getSeconds();
+                          if (dsInterval < rollupInterval) {
```

Comment:
And we need a validation that the rollup interval is a multiple of the source interval. E.g. if they give us a 67 second downsample we should throw.